coffee_core/src/coffee.rs

```diff
@@ -597,6 +597,68 @@ impl PluginManager for CoffeeManager {
             .await?;
         Ok(tip)
     }
+
+    async fn disable(&mut self, plugin: &str) -> Result<(), CoffeeError> {
+        log::debug!("disabling plugin: {plugin}");
+
+        let plugin = self
+            .config
+            .plugins
+            .iter_mut()
+            .find(|repo_plugin| plugin == repo_plugin.name())
+            .ok_or(error!(
+                "No plugin with name `{plugin}` found in the plugins installed"
+            ))?;
+        log::debug!("plugin: {:?}", plugin);
+        if plugin.enabled == Some(false) {
+            return Err(error!("Plugin `{plugin}` is already disabled"));
+        }
+        self.coffee_cln_config
+            .rm_conf("plugin", Some(&plugin.exec_path))
+            .map_err(|err| error!("{}", err.cause))?;
```

Comment:
nit: cln has an option that it is `disable-plugin=PLUGIN` with the following docs

>If PLUGIN contains a /, plugins with the same path as PLUGIN will
not be loaded at startup. Otherwise, no plugin with that base name will
be loaded at startup, whatever directory it is in. This option is useful for
disabling a single plugin inside a directory. You can still explicitly
load plugins which have been disabled, using lightning-plugin(7) start.

Reply:
Yes that's much better
Thanks